service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -189,13 +188,15 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
     // endpoint
     String rawMethodUrl;
     try {
+      PostMethodRequest.MethodSourceEnum methodSourceEnum =
+          PostMethodRequest.MethodSourceEnum.fromValue(methodVersion.method().methodSource());
+
+      if (methodSourceEnum == null) {
+        throw new UnknownMethodSourceException(methodVersion.method().methodSource());
+      }
       rawMethodUrl =
           MethodUtil.convertToRawUrl(
-              methodVersion.url(),
-              Objects.requireNonNull(
-                  convertToMethodSourceEnum(methodVersion.method().methodSource())),
```

Comment:
a bad `MethodSource` field was the root cause of most of the verification failures. It showed up as a vague null pointer exception, which was thrown by the above `requireNonNull` call.

I'm replacing this `requireNonNull` with the `if` statement above that throws the more specific `UnknownMethodSourceException`

Reply:
we need to preserve the use of `convertToMethodSource`

Reply:
What happened to the explicit `UnknownMethodSourceException` throw?

Reply:
turns out `convertToMethodSource` throws it! I didn't realize that before